src/Model/Filter/Boolean.php

```diff
@@ -25,7 +25,12 @@ public function process()
             return;
         }
 
-        $value = strtolower($this->value());
+        $value = $this->value();
+        if (!is_scalar($value)) {
+            return;
+        }
+
+        $value = strtolower($value);
```

Comment:
Ah, didn't saw that this was a new PR @inoas, I've pushed tests and a fix too :)